library/core/src/slice/mod.rs

```diff
@@ -641,11 +653,23 @@ impl<T> [T] {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn iter(&self) -> Iter<'_, T> {
-        // SAFETY: adding `self.len()` to the starting pointer gives a pointer
-        // at the end of `self`, which fulfills the expectations of `ptr.add()`
-        // and `NonNull::new_unchecked()`.
+        let ptr = self.as_ptr();
+        // SAFETY: There are several things here:
+        //
+        // `ptr` has been checked for nullity before being passed to `NonNull` via
+        // `new_unchecked`.
```

Comment:
Nit: it hasn't really been "checked" in this method right? I think that's just a guarantee by the representation of slices that `self.as_ptr()` is never null.

Reply:
I kept reading `assume` as `assert` for some reason, you're completely right